sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/MultipartFormDataHelper.java

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) TypeSpec Code Generator.
+
+package com.azure.ai.openai.implementation;
+
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+// DO NOT modify this helper class
+
+public final class MultipartFormDataHelper {
```

Comment:
Before the removing this implementation I would perform a `LIVE` test, to be sure that everything still works as expected. WDYT, @mssfang ?

Reply:
We can leave it in the `implementation` folder before we remove our hand-written multpart/form-data APIs (which also hide internally class/methods). I have tried to play with this new model for a bit. But still can't get it work.

Replaced hand-written code
```java
        final MultipartDataHelper helper = new MultipartDataHelper();
        final MultipartDataSerializationResult result = helper.serializeRequest(audioTranscriptionOptions);
        final BinaryData data = result.getData();
        requestOptions = helper.getRequestOptionsForMultipartFormData(requestOptions, result, helper.getBoundary());
```
by using this class implementation,

```java
        requestOptions = requestOptions == null ? new RequestOptions() : requestOptions;
        final MultipartFormDataHelper helper = new MultipartFormDataHelper(requestOptions);
        List<MultipartField> multipartFields = formatAudioTranscriptionOptions(audioTranscriptionOptions);
//        helper.serializeFileField("file", BinaryData.fromBytes(audioTranscriptionOptions.getFile()), fileName);

        for (MultipartField field : multipartFields) {
            helper.serializeFileField(field.getWireName(), BinaryData.fromBytes(audioTranscriptionOptions.getFile()), fileName);
        }
        helper.end();
        BinaryData data = helper.getRequestBody();

```



Error:

```
com.azure.core.exception.HttpResponseException: Status code 400, "{
  "error": {
    "message": "2 validation errors for Request\nbody -> file\n  field required (type=value_error.missing)\nbody -> response_format\n  value is not a valid enumeration member; permitted: 'json', 'text', 'vtt', 'srt', 'verbose_json' (type=type_error.enum; enum_values=[<ResponseFormat.JSON: 'json'>, <ResponseFormat.TEXT: 'text'>, <ResponseFormat.VTT: 'vtt'>, <ResponseFormat.SRT: 'srt'>, <ResponseFormat.VERBOSE_JSON: 'verbose_json'>])",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}"
```
With request payload for original code:

```
--31b586c2-35db-48
Content-Disposition: form-data; name="response_format"; filename="batman.wav"
Content-Type: application/octet-stream

// ...

--31b586c2-35db-48--

```

Request payload by using the new implementation.

```
--6512c210-932b-4a
Content-Disposition: form-data; name="file"; filename="batman.wav"
Content-Type: application/octet-stream

// ...

--6512c210-932b-4a
Content-Disposition: form-data; name="response_format"

json
--6512c210-932b-4a--
```


Maybe how I use the new implementation is wrong.

Reply:
We likely won't switch to this `MultipartFormDataHelper` impl any time soon (as it handles stream, but not handling replayable correctly -- this requires improvement on core). All the changes in this PR is just to keep using existing code (aka `MultipartDataHelper`), not changing a bit.

---

The code above is not correct. For the new helper, it won't need call to `formatAudioTranscriptionOptions` (and in the line after, you are using the wireName from its output with the file data, which is not match).

The expected code (if all using new helper) would be something like this
https://github.com/Azure/azure-sdk-for-java/blob/55fb5b37807d8df75942fe31c2a99c9431b4399a/sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java#L1603-L1620

But need to modify it a bit, remove the "filename" field as it is not a field, and modify the file to use `byte[]` etc.
```java
        RequestOptions requestOptions = new RequestOptions();
        return getAudioTranscriptionAsResponseObjectWithResponse(deploymentOrModelName,
            new MultipartFormDataHelper(requestOptions)
                .serializeFileField("file", BinaryData.fromBytes(audioTranscriptionOptions.getFile()),
                    audioTranscriptionOptions.getFilename())
                .serializeTextField("response_format", Objects.toString(audioTranscriptionOptions.getResponseFormat()))
                .serializeTextField("language", audioTranscriptionOptions.getLanguage())
                .serializeTextField("prompt", audioTranscriptionOptions.getPrompt())
                .serializeTextField("temperature", Objects.toString(audioTranscriptionOptions.getTemperature()))
                .serializeTextField("model", audioTranscriptionOptions.getModel()).end().getRequestBody(),
            requestOptions).getValue().toObject(AudioTranscription.class);
```

I actually tested it on OpenAI as final test for this feature in emitter (above code is just a demonstration, I've reverted my test code).

Reply:
> Before the removing this implementation I would perform a `LIVE` test, to be sure that everything still works as expected. WDYT, @mssfang ?

Maybe a misunderstanding. This class is what added by emitter for the generated code serializing data for multipart.
I mean it is NOT USED, because OpenAI already doing this via existing `MultipartDataHelper` class. And I don't want to change that a bit.

All this is for new service that won't have the people to write all these customization in OpenAI (we had pending Purview Datamap, one from Vision, one from Face API that having multipart in their typespec).

Reply:
We can remove this file until this `MultipartFormDataHelper` is ready to use.